### ISSUE TITLE
Too many open files error on latest prometheus version

### ISSUE THREAD

**Issue by vasukiprasad1** [prometheus/prometheus#6804]
After upgrading the prometheus operator from 0.29.0 to 0.35.0 and also upgraded the prometheus from v2.8.0 to v2.15.2.  Prometheus pods are crashing while the load test is on. We had ran the similar test on the older version with similar HW underneath, load test was successful.

```
level=info ts=2020-02-10T16:05:34.380Z caller=main.go:617 msg="Server is ready to receive web requests."
ts=2020-02-10T16:05:42.439Z caller=dedupe.go:111 component=remote level=info remote_name=a40741 url=http://localhost:9095/receive msg="done replaying WAL" duration=8.074058943s
ts=2020-02-10T16:06:00.429Z caller=dedupe.go:111 component=remote level=error remote_name=a40741 url=http://localhost:9095/receive msg="error tailing WAL" err="segments: open /prometheus/wal: too many open files"
ts=2020-02-10T16:06:05.429Z caller=dedupe.go:111 component=remote level=error remote_name=a40741 url=http://localhost:9095/receive msg="error tailing WAL" err="wal.Segments: open /prometheus/wal: too many open files"
level=warn ts=2020-02-10T16:06:33.470Z caller=main.go:494 msg="Received SIGTERM, exiting gracefully..."
level=info ts=2020-02-10T16:06:33.470Z caller=main.go:517 msg="Stopping scrape discovery manager..." 
(edited)
```

Startup logs from older prometheus version.
```level=info ts=2020-01-15T00:12:14.68377516Z caller=main.go:324 fd_limits="(soft=1048576, hard=1048576)```

Startup logs on newer prometheus version.
```level=info ts=2020-02-10T20:29:23.101Z caller=main.go:333 fd_limits="(soft=1024, hard=4096)"```

Looks like this is a bug where prometheus is not setting the correct fd ulimit. 

Can someone let us know how we can increase both soft and hard limit for prometheus image on start up.

**Comment by beorn7** [comment]
AFAIK Prometheus doesn't even try to change the fd limit. This smells more like a changed setting in your environment (or perhaps in the Prometheus operator).

**Comment by vasukiprasad1** [comment]
Thanks for your reply @beorn7. I will downgrade the operator and keep the latest Prometheus and test it.

I will keep this thread updated.

**Comment by vasukiprasad1** [comment]
We found the problem, latest kops are updating the docker version and also installing some external dependencies which are configuring the default configuration for docker containers in EC2 instance which are setting the lower fd_limit value. We disabled the external dependencies installation option and ran the test and we got expected fd_limits. 

This issue can be closed.

Thanks for the help.
